### PR TITLE
[gsoap]Add new port

### PR DESCRIPTION
**Pull request by JackBoosY**
For some reason, the source code does not suitable to build gsoap binary files. So I use binary files in source path instead of build.
see #4457

### CODE REVIEW

**Comment by martin-s** [comment]
I would rather like to see the source build instead of the binaries. Build should be working as I have built it in the past. Can you name the reason it's failing?

**Comment by JackBoosY** [comment]
@martin-s  I  will check build cfg again.thanks.

**Comment by JackBoosY** [comment]
Due to a problem with the implementation of the patch[#5380], file[copy] is now used to make patch with .sln file.

File: ports/gsoap/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+else()
+    message("gsoap only supported Win32")
+    set(BUILD_ARCH "Win32")
+endif()
+
+# Handle binary files and includes
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/gsoap ${CURRENT_PACKAGES_DIR}/debug/tools)
+
+if (WIN32)
+    file(COPY ${CURRENT_PORT_DIR}/soapcpp2.sln DESTINATION ${SOURCE_PATH}/gsoap/VisualStudio2005/soapcpp2/)
```
Comment (Rastaban) [inline ports/gsoap/portfile.cmake]: Could you add a comment here pointing out that the patch failed so you copied the files as suggested by issue #442.

File: ports/gsoap/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://ayera.dl.sourceforge.net/project/gsoap2/gsoap-${GSOAP_VERSION}/gsoap_${GSOAP_VERSION}${GSOAP_SUB_VERSION}.zip"
+    FILENAME "gsoap_${GSOAP_VERSION}${GSOAP_SUB_VERSION}.zip"
+    SHA512 c115044d2662c2dd355c4756a974a0013b7213dd28c536aba179e53c19466279bfa34ce16b4426db5aa7a24d94c18e0ed7e7cdf05e799bf89f7b54031aa0874e
+)
+
+vcpkg_extract_source_archive(${ARCHIVE})
+
+#vcpkg_extract_source_archive_ex(
```
Comment (Rastaban) [inline ports/gsoap/portfile.cmake]: Remove the commented out code

**Comment by Rastaban** [comment]
This looks good to me.  The only build failures in testing I see are linux/osx which are by design.  After you respond to my minor review comments I can merge the PR.

**Comment by martin-s** [comment]
You can use the patch file here:
[fix-build-in-windows.zip](https://github.com/Microsoft/vcpkg/files/2895620/fix-build-in-windows.zip)

We could add also a 64-bit port, but this no blocker.

**Comment by JackBoosY** [comment]
@Rastaban  I have no able to build gsoap in Linux/OSX, maybe someone can help me?

**Comment by JackBoosY** [comment]
> You can use the patch file here:
> [fix-build-in-windows.zip](https://github.com/Microsoft/vcpkg/files/2895620/fix-build-in-windows.zip)
> 
> We could also add a 64-bit port, but this is no blocker.

Patch works! Thanks. I have no idea why my patch applying failed.

**Comment by Rastaban** [comment]
@JackBoosY thank you for your work! I won't hold up this PR for the OSX and Linux build failures because having this library in and working for the other triplets is better than not having it in at all.  If we can find someone interested in getting it working for OSX/Linux then we can do it in a separate PR.

**Pull request merged**